cmd/kubent/main.go

```diff
@@ -119,18 +119,8 @@ func main() {
 		log.Fatal().Err(err).Str("name", "Rego").Msg("Failed to evaluate input")
 	}
 
-	var prnt printer.Printer
-	switch config.Output {
-	case "json":
-		prnt, err = printer.NewJSONPrinter(&printer.JSONOpts{})
-	default:
-		prnt, err = printer.NewTextPrinter(&printer.TextOpts{})
-	}
-	if err != nil {
-		log.Fatal().Err(err).Str("name", config.Output).Msg("Failed to initialize output printer")
-	}
-
-	err = prnt.Print(results)
+	printer, _ := printer.ParsePrinter(config.Output)
+	err = printer().Print(results)
```

Comment:
It would be a bit nicer to save the printer, and then call `Print` on it, this feels a bit too much at once + not saving the printer anywhere.

Reply:
Thinking about this a bit more, it might be better to offload this to a separate fn in the printer package (not necessary, just to keep main code clean) and combine it with the previous, e.g.:
```go
printer, err := printer.newPrinter(config.Output)
if err...
...
err = printer.Print(results)
```